### PR TITLE
Rename `metaflow_custom` to `metaflow_extensions`

### PR DESCRIPTION
**Pull request by savingoyal**

### CODE REVIEW

File: metaflow/__init__.py [hunk 1]
```diff
@@ -219,43 +219,43 @@ def _can_handle_orig_module():
...
 except ImportError as e:
     ver = sys.version_info[0] * 10 + sys.version_info[1]
     if ver >= 36:
         # e.name is set to the name of the package that fails to load
         # so don't error ONLY IF the error is importing this module (but do
         # error if there is a transitive import error)
         if not (isinstance(e, ModuleNotFoundError) and \
-                e.name in ['metaflow_custom', 'metaflow_custom.toplevel',
-                           'metaflow_custom.toplevel.toplevel']):
+                e.name in ['metaflow_extensions', 'metaflow_extensions.toplevel',
```
Comment (savingoyal) [inline metaflow/__init__.py:230]: @romain-intel This change may require a corresponding change in the nflx-internal metaflow-extensions.

File: metaflow/__init__.py [hunk 2]
```diff
@@ -219,43 +219,43 @@ def _can_handle_orig_module():
...
         elif isinstance(o, types.ModuleType) and o.__package__ and \
-                o.__package__.startswith('metaflow_custom'):
+                o.__package__.startswith('metaflow_extensions'):
             lazy_load_custom_modules['metaflow.%s' % n] = o
     if lazy_load_custom_modules:
         # Prepend to make sure custom package overrides things
         sys.meta_path = [_LazyLoader(lazy_load_custom_modules)] + sys.meta_path
 
-    __version_addl__ = getattr(extension_module, '__mf_customization__', '<unk>')
+    __version_addl__ = getattr(extension_module, '__mf_extensions__', '<unk>')
```
Comment (savingoyal) [inline metaflow/__init__.py:258]: @romain-intel This change may require a corresponding change in the nflx-internal metaflow-extensions.

File: metaflow/__init__.py (hunk 1 above)
Comment (romain-intel) [reply metaflow/__init__.py:230]: There is one missing there (the toplevel.toplevel). OK for nflx package.

File: metaflow/__init__.py (hunk 2 above)
Comment (romain-intel) [reply metaflow/__init__.py:258]: done

File: metaflow/__init__.py (hunk 1 above)
Comment (romain-intel) [reply metaflow/__init__.py:230]: oh nvm, it was below.

**Review by romain-intel — APPROVED** [review]
LGTM *except* that the test/core/metaflow_custom should be updated properly (changed to test/core/metaflow_extensions and files in there may need a few updates if needed. Please fix that (see https://github.com/Netflix/metaflow/commit/a4823a57cdb6fd8c05c72006558aea273e4b6a54)

**Comment by savingoyal** [comment]
> LGTM _except_ that the test/core/metaflow_custom should be updated properly (changed to test/core/metaflow_extensions and files in there may need a few updates if needed. Please fix that (see [a4823a5](https://github.com/Netflix/metaflow/commit/a4823a57cdb6fd8c05c72006558aea273e4b6a54))

Done!

**Comment by savingoyal** [comment]
I will merge this PR once the tests pass.

**Comment by savingoyal** [comment]
@bishax You may want to make note of this change. Just renaming `custom` to `extensions` should work.

**Pull request merged**